docs/source/usage_guides/gradient_accumulation.md

```diff
@@ -118,8 +118,16 @@ You can remove all the special checks for the step number and the loss adjustmen
 As you can see the [`Accelerator`] is able to keep track of the batch number you are on and it will automatically know whether to step through the prepared optimizer and how to adjust the loss. 
 
 <Tip>
+
 Typically with gradient accumulation, you would need to adjust the number of steps to reflect the change in total batches you are 
-training on. 🤗 Accelerate automagically does this for you by default. Behind the scenes we instantiate a GradientAccumulationPlugin configured to do this.
+training on. 🤗 Accelerate automagically does this for you by default. Behind the scenes we instantiate a [`GradientAccumulationPlugin`] configured to do this.
+
+</Tip>
+
+<Tip warning={true}>
+
+The [`state.GradientState`] (what runs this code and how Accelerate knows how to do everything) is sync'd with the active dataloader being iterated upon. As such it assumes naively that when we have reached the end of the dataloader everything will sync and a step will be performed. To disable this, set `sync_with_dataloader` to be `False` in the [`GradientAccumulationPlugin`].
```

Comment:
> (what runs this code and how Accelerate knows how to do everything)

Not sure how helpful this info is for the reader, I think it can be deleted.

> To disable this, set `sync_with_dataloader` to be `False` in the [`GradientAccumulationPlugin`]

Maybe explicitly write the code for this, like

`accelerator = Accelerator(..., gradient_accumulation_plugin=GradientAccumulationPlugin(sync_with_dataloader=False))`

unless you think it's obvious to the average user.